### PR TITLE
Design pattern

### PR DESCRIPTION
**Pull request by Javivy**
## In this milestone I implemented the following functions and enhancements:

  - [x] Made use of Decorator Design Pattern
  - [x] Created a decorator class 
  - [x] Created a Capitalize Decorator that inherits from the Decorator class
  - [x] Created a Trimmer Decorator that inherits from the Decorator class

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,10 +27,9 @@ After you're finished please remove all the comments and instructions!
 
 <div align="center">
   <!-- You are encouraged to replace this logo with your own! Otherwise you can also remove it. -->
-  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
   <br/>
 
-  <h3><b>Microverse README Template</b></h3>
+  <h3><b>School Library</b></h3>
```
Comment (CollinsTatang) [inline README.md]: - [ ] Please update your README file following the [Microverse README Template](https://github.com/microverseinc/readme-template)
- [ ] Please do not put the project name in the position of the logo.

- [ ] Please remove the microverse logo **Optional** You can put yours if you have one.

**Review by CollinsTatang — COMMENTED** [review]
Hi @Javivy, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that required an updated README file.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 2]
```diff
@@ -1,14 +1,23 @@
-class Person
+require './nameable'
+require './capitalize_decorator'
+require './trimmer_decorator'
+
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
   def initialize(age, name: 'Unknown', parent_permission: true)
```
Comment (Davidosky007) [inline person.rb]: * I get an error here when I try to test your decoration, I think the issue is that you are setting the name argument to unknown so I noticed when I changed it to ` name = "unknown" it worked. check the screenshots below ⬇️ 
* Also I noticed your trimmer method isn't functioning as expected as you can see in the screenshot the word was not trimmed , Kindly work on these 🙏 
image
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Javivy 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 
so-close

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.delete(' ')
+  end
```
Comment (mahbubul14) [inline trimmer_decorator.rb]: - [ ] As the previous reviewer suggested, Kindly work on the trimmer method. :+1: 

> **Project requirement:** Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @Javivy,

Great work on making the changes requested by the previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @Javivy , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights :green_circle: 

- Great commit history :heavy_check_mark: 
- All Linter checks passed :heavy_check_mark: 
- PR is professional :heavy_check_mark: 

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**